File: server/src/model/StorageItem.java
package model;

import util.database.DataModel;

public class StorageItem extends DataModel {
    
    private String typeItem;
//    private String []resourceItem;
    private int quantity;
    
    public StorageItem(String typeItem, int quantity) {
        super();
        
        this.typeItem = typeItem;
//        this.resourceItem = resourceItem;
        this.quantity = quantity;
    }

    
    public String getTypeItem() {
        return typeItem;
    }

    public int getQuantity() {
        return quantity;
    }
    
    
    
    public boolean addQuantity(int value){
        this.quantity += value;
        return true;
    }
    public boolean reduceQuantity(int value){
        if (this.quantity >= value){
            this.quantity -= value;
            return true;
        }
        return false;
    }
}

File: client/src/modules/models/StorageItem.js

var StorageItem = cc.Class.extend({

    typeItem: null,
    quantity: 0,

    ctor: function (typeProduct, quantity) {
        //
        //this._super();

    //     this.render(typeProduct, quantity);

    // },
    // render: function (typeProduct, quantity) {

        this.typeItem = typeProduct;
        this.quantity = quantity;
    },

    addQuantity: function (number) {
        this.quantity += number;

    },
    reduceQuantity: function (number) {
        this.quantity -= number;
    },

    getTypeItem: function () {
        return this.typeItem;
    },
    getQuantityItem: function () {
        return this.quantity;
    },

    //
    getLevelUnlock: function () {
        return getProductObjByType(this.typeItem).level;
    }

});